### ISSUE TITLE
[Search Query] Better regex support

### ISSUE THREAD

**Issue by taiyab** [sourcegraph/sourcegraph#58428]
When using a regex as part of my search query, I'm not quite sure whether the system has recognised my input as a regex.

I also wouldn't know a `Enable regular expression` option had to be switched on.

<img width="1491" alt="image" src="https://github.com/sourcegraph/sourcegraph/assets/5337876/8c2ef20a-4500-4aa3-b640-d85a5525257c">

**Comment by keegancsmith** [comment]
Right now hitting the regex button disabled keyword search and uses our old style which concats the tokens with `.*`. I think we discussed removing it, but we punted on that for this quarter. We know we have some customers/power users which love this mode.

**Comment by jtibshirani** [comment]
I wonder if we could remove the button and instead surface `patternType:regexp` in the search bar itself. This would simplify the UX for new and casual users, while still supporting the "power user" use case.

An example of what I've seen from power users: they configure a "custom search engine" in their browser like `https://sourcegraph.com/search?q=julie&patternType=regexp`.

**Comment by camdencheek** [comment]
> instead surface patternType:regexp in the search bar itself.

FWIW, this already works. If you type `patterntype:regexp` in the search bar, it will just remove it and activate the button.

**Comment by jtibshirani** [comment]
> We could remove that behavior and maintain backwards compatibility by just leaving patterntype:regexp in the query text

Yes, that's what I was referring to -- leaving it in the query text. (My wording was just not super clear)
